src/Tes.Runner/Docker/DockerExecutor.cs

```diff
@@ -16,20 +15,23 @@ namespace Tes.Runner.Docker
 {
     public class DockerExecutor
     {
-
         private readonly IDockerClient dockerClient = null!;
         private readonly ILogger logger = PipelineLoggerFactory.Create<DockerExecutor>();
-        private readonly NetworkUtility networkUtility = new NetworkUtility();
-        private readonly RetryHandler retryHandler = new RetryHandler(Options.Create(new RetryPolicyOptions()));
+        private readonly NetworkUtility networkUtility = new();
+        private readonly Polly.Retry.AsyncRetryPolicy asyncRetryPolicy = null!;
         private readonly IStreamLogReader streamLogReader = null!;
         private readonly ContainerRegistryAuthorizationManager containerRegistryAuthorizationManager = null!;
+        private readonly Predicate<DockerApiException> IsAuthFailure = e => !IsNotAuthFailure(e);
+        private static readonly Func<DockerApiException, bool> IsNotAuthFailure = e =>
+            !(e.StatusCode == System.Net.HttpStatusCode.Unauthorized || e.StatusCode == System.Net.HttpStatusCode.Forbidden ||
+                !(e.StatusCode == System.Net.HttpStatusCode.InternalServerError &&
+                    new[] { "unauthorized", "forbidden" }.Contains(e.ResponseBody?.Split(':').Skip(1).FirstOrDefault()?.ToLower() ?? string.Empty)));
```

Comment:
Can you add a comment that explains this in plain English? Also ToLowerInvariant()